.github/workflows/build.yml

```diff
@@ -31,6 +31,25 @@ jobs:
       with:
         go-version: ^1.16.1
       id: go
+    - name: Cache modules and build
+      uses: actions/cache@v3
+      with:
+        # In order:
+        # * Module download cache
+        # * Build cache (Linux)
+        # * Build cache (Mac)
+        # * Build cache (Windows)
+        path: |
+          ~/go/pkg/mod
+          ~/.cache/go-build
+          ~/Library/Caches/go-build
+          %LocalAppData%\go-build
+        key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-go-
+    - name: Download modules
+      if: steps.cache.outputs.cache-hit != 'true'
+      run: go mod download
```

Comment:
Some tests on github actions are randomly failed because of timed out of test execution.
I tried to fix them but I didn't find out the root cause, and even now I don't know any solution.

My trials were extending test executing time and caching dependencies, and unfortunately both of them were ineffective.
But I think caching is valuable so the cache setting is remained.